src/Microsoft.TemplateEngine.Edge/Constraints/OSConstraint.cs

```diff
@@ -16,6 +17,13 @@ namespace Microsoft.TemplateEngine.Edge.Constraints
 {
     internal class OSConstraintFactory : ITemplateConstraintFactory
     {
+        private static readonly Dictionary<string, OSPlatform> _platformMap = new Dictionary<string, OSPlatform>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "Windows",  OSPlatform.Windows },
+            { "Linux",  OSPlatform.Linux },
+            { "OSX",  OSPlatform.OSX }
+        };
```

Comment:
I'm thinking whether `OSPlatform` is good enough for target use cases.
Looking into maui codebase, they do not use it at all (outside of some tests). In their codnitions (e.g. [`OnPlatfrom`](https://github.com/dotnet/maui/blob/f2a42f6aadd018144f191ccb6107734f0904513a/src/Controls/src/Core/OnPlatform.cs#L8)), they rely on [`Device`](https://github.com/dotnet/maui/blob/f2a42f6aadd018144f191ccb6107734f0904513a/src/Controls/src/Core/Device.cs) or it's replacement [`DevicePlatform`](https://github.com/dotnet/maui/blob/4b7dd8acaf2b0040a6a07e71fc140505ee68b7ac/src/Essentials/src/Types/DevicePlatform.shared.cs) (which misses GTK and WPF, but there is probably good reason for that).

**tl;dr;:** We should probably use same level of categorization as Maui codebase or if we use lower level of detail then we should check with them if it's good enough for their use cases.

Reply:
I would avoid building on top of MAUI classes (in general).
`OperatingSystem` seems to be a better alternative (https://docs.microsoft.com/en-us/dotnet/api/system.operatingsystem?f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(System.OperatingSystem)%3Bk(SolutionItemsProject)%3Bk(DevLang-csharp)%26rd%3Dtrue&view=net-6.0)

Reply:
After online talk with @vlada-shubina, it seems that `OSPlatform` should actually suffice here - as we are looking at design time constraint, not a run time constraint.